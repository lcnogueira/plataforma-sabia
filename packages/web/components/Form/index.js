export { default as InputField } from './InputField';
export { default as SwitchField } from './SwitchField';
export { default as TextField } from './TextField';
export { default as SelectField } from './SelectField';
export { default as CheckBoxField } from './CheckBoxField';
export { default as MaskedInputField } from './MaskedInputField';
export { default as CurrencyInputField } from './CurrencyInputField';
export { default as Watcher } from './Watcher';
export { default as InputHiddenField } from './InputHiddenField';
export { default as Repeater } from './Repeater';
export { default as Help } from './Help';
export { default as HelpModal } from './HelpModal';
export * from './Form';
