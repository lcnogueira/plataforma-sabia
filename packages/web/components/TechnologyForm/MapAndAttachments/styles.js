import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
	margin-bottom: 4rem;
`;

export const UploadBox = styled.div`
	flex-direction: row;
	display: flex;
	margin: 0;
	margin-bottom: 1rem;
	border: 1px solid ${({ theme }) => theme.colors.lightGray3};
	padding: 1rem;
	cursor: pointer;

	&:hover {
		background-color: ${({ theme }) => theme.colors.lightGray3};
	}
`;

export const IconLink = styled.a`
	color: ${({ theme }) => theme.colors.darkGray};
`;

export const Place = styled.div``;

export const Div = styled.div``;

export const IconBox = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;

	& > * {
		margin: 0.25rem;
	}
`;

export const IconTextBox = styled.div`
	font-size: 1.2rem;
	font-weight: 300;
	display: flex;
	justify-content: flex-start;
	align-items: flex-start;
	flex-direction: column;
`;

export const InputTextBox = styled.div`
	flex: 3;
	margin: auto 0;
`;

export const UploadedDocuments = styled.div`
	display: flex;
	flex-direction: column;
`;

export const UploadedImages = styled.div`
	display: grid;
	grid-template-columns: repeat(auto-fill, 30%);
	grid-gap: 1rem;

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		grid-template-columns: repeat(auto-fill, 100%);
	}
`;

export const Suggestion = styled.div`
	margin: 0.5rem 0;
	padding: 1rem;
	width: 100%;
	font-size: 1.4rem;
	cursor: pointer;
`;

export const GoogleAddressSugestions = styled.div`
	display: flex;
	flex-direction: column;
`;

export const SvgBox = styled.div`
	text-align: center;
	flex: 1;
	margin: auto 0;
`;

export const Title = styled.h3``;

export const Media = styled.img`
	width: 100%;
	filter: opacity(0.7);
`;

export const IconRow = styled.div`
	${({ row, alignItems }) => css`
		display: flex;
		flex-direction: ${row ? 'row' : 'column'};

		& > * {
			margin-bottom: 1rem;
		}

		& > a {
			padding: 0;
			margin-right: 1rem;
			display: flex;
		}

		align-items: ${alignItems || 'center'};
	`}
`;

export const Button = styled.button.attrs(() => ({ type: 'button' }))`
	${({ theme: { colors } }) => css`
		border: none;
		outline: none;

		display: flex;
		align-items: center;

		padding: 0.2rem;
		padding-left: 0.6rem !important;
		padding-right: 0.6rem !important;
		line-height: 2.4rem;

		text-transform: uppercase;
		font-size: 1.4rem;
		font-weight: 700;
		color: ${colors.red};
		border: 2px solid ${colors.red};

		> svg {
			margin-right: 0.4rem;
		}

		:hover,
		:focus {
			color: ${colors.white};
			background: ${colors.red};
		}
	`}
`;

export const RadioWrapper = styled.div`
	${({ theme: { colors } }) => css`
		display: flex;
		align-items: center;
		white-space: nowrap;

		> input {
			appearance: none;
		}
		> input:checked + label {
			color: ${colors.white};
			background-color: ${colors.secondary};
			border: 2px solid ${colors.secondary};
		}
		> label {
			display: flex;
			align-items: center;
			color: ${colors.lightGray2};
			text-align: center;
			padding: 0.2rem 0.6rem;
			cursor: pointer;
			font-size: 1.4rem;
			font-weight: 700;
			line-height: 2.4rem;
			border: 2px solid ${colors.lightGray2};
			text-transform: uppercase;

			> svg {
				margin-right: 0.4rem;
			}
		}
		:focus-within > label {
			background-color: ${colors.secondary};
		}
	`}
`;

export const InputVideoWrapper = styled.section`
	width: 100%;
	display: flex;
	align-items: center;
	margin: 0 -8px 16px;

	> div {
		flex: 1;
		margin-right: 10px;
	}
`;

const VideoContainer = styled.section`
	background: ${({ theme: { colors } }) => colors.white};
	width: 100%;
	padding: 16px 32px 32px;
	margin-top: 8px;
	margin-bottom: 16px;
`;

const VideosWrapper = styled.section`
	border-top: 1px solid ${({ theme: { colors } }) => colors.lightGray4};
	border-bottom: 1px solid ${({ theme: { colors } }) => colors.lightGray4};
	padding: 16px 16px 0;
	margin: 0 -32px;
	display: flex;
	flex-wrap: wrap;
	align-items: center;
	flex-wrap: wrap;
`;

const VideoItem = styled.section`
	width: 130px;
	display: flex;
	flex-direction: column;
	margin: 0 8px 16px;

	a {
		display: block;
		padding: 0;
		margin-bottom: 10px;

		&:hover {
			opacity: 0.7;
		}
	}

	img {
		background: ${({ theme: { colors } }) => colors.lightGray2};
		width: 100%;
		display: block;
	}
`;

const RemoveVideoButton = styled.button`
	display: flex;
	align-items: center;
	align-self: center;
	border: 0;
	outline: none;

	text-transform: uppercase;
	font-weight: bold;
	font-size: 1.4rem;
	line-height: 2.4rem;

	background: none;
	color: ${({ theme: { colors } }) => colors.red};
	padding: 0.2rem 0.6rem;

	:hover,
	:focus {
		color: ${({ theme: { colors } }) => colors.white};
		background: ${({ theme: { colors } }) => colors.red};
	}
`;

const EmptyVideos = styled.section`
	text-align: center;
`;

export const Videos = ({ data, onRemove, children }) => (
	<VideoContainer>
		{children}
		{data?.length ? (
			<VideosWrapper>
				{data.map((video, idx) => (
					<VideoItem key={`video_${video.videoId}`}>
						<a
							href={`//www.youtube.com/watch?v=${video.videoId}`}
							target="_blank"
							rel="noreferrer"
						>
							<img src={video.thumbnail} alt={`Youtube video ${video.videoId}`} />
						</a>
						<RemoveVideoButton type="button" onClick={() => onRemove(idx)}>
							Remover
						</RemoveVideoButton>
					</VideoItem>
				))}
			</VideosWrapper>
		) : (
			<EmptyVideos>
				<p>Nenhum video adicionado</p>
			</EmptyVideos>
		)}
	</VideoContainer>
);

Videos.propTypes = {
	data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
	children: PropTypes.children().isRequired,
	onRemove: PropTypes.func().isRequired,
};
