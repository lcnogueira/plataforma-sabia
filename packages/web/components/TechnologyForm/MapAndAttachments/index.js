/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-use-before-define */
import React, { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { FaMinus, FaPlus, FaTrash, FaFileUpload, FaFilePdf, FaMapMarkerAlt } from 'react-icons/fa';
import Dropzone from 'react-dropzone';
import CreatableSelect from 'react-select/creatable';
import PlacesAutocomplete, { geocodeByPlaceId } from 'react-places-autocomplete';
import { InputField, SelectField } from '../../Form';
import {
	UploadedImages,
	UploadedDocuments,
	SvgBox,
	IconBox,
	IconTextBox,
	UploadBox,
	Suggestion,
	GoogleAddressSugestions,
	Div,
	IconLink,
	Wrapper,
	IconRow,
	Title,
	Media,
} from './styles';
import { Row, Column } from '../../Common/Layout';
import { CircularButton } from '../../Button';
import Repeater from '../../Form/Repeater';

const MapAndAttachments = ({ form }) => {
	const imgsRef = useRef(null);
	const pdfRef = useRef(null);
	const [addressInput, setAddressInput] = useState([]);
	const [address, setAddress] = useState([]);
	const [previewedImgFiles, setPreviewedImgFiles] = useState([]);
	const [previewedPdfFiles, setPreviewedPdfFiles] = useState([]);
	// eslint-disable-next-line consistent-return
	const onDropImgs = (acceptedFiles) => {
		if (!acceptedFiles) return null;
		const parsedJsx = [];

		for (let index = 0; index < acceptedFiles.length; index += 1) {
			parsedJsx.push(acceptedFiles[index]);
		}
		setPreviewedImgFiles([...previewedImgFiles, ...parsedJsx]);
	};

	// eslint-disable-next-line consistent-return
	const onDropPdfs = (acceptedFiles) => {
		if (!acceptedFiles) return null;
		const parsedJsx = [];

		for (let index = 0; index < acceptedFiles.length; index += 1) {
			parsedJsx.push(acceptedFiles[index]);
		}
		setPreviewedPdfFiles([...previewedPdfFiles, ...parsedJsx]);
	};

	const deleteAttachment = (index) => {
		setPreviewedPdfFiles(
			previewedPdfFiles.filter((element, innerIndex) => index !== innerIndex),
		);
	};

	const deleteMedia = (index) => {
		setPreviewedImgFiles(
			previewedImgFiles.filter((element, innerIndex) => index !== innerIndex),
		);
	};

	const deleteAlreadyImpletedMarker = (index) => {
		setAddress(address.filter((element, innerIndex) => index !== innerIndex));
	};

	return (
		<Wrapper>
			<Row>
				<Column>
					<Row>
						<Column>
							<Title>Onde a Tecnologia é desenvolvida?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<Repeater
								form={form}
								name="who_develop"
								childsComponent={({ remove, fields, index }) => {
									return (
										<Row align="center" justify="center">
											<Column noPadding>
												<CreatableSelect
													form={form}
													name={`who_develop[${index}]`}
													placeholder="Nome da instituição que desenvolve a tecnologia"
													validation={{ required: true }}
													options={[
														{
															value: 'INSA',
															label:
																'INSTITUTO NACIONAL DO SEMIÁRIDO',
														},
													]}
												/>
											</Column>
											<Column autoX>
												<CircularButton
													disabled={index === 0 && fields.length === 1}
													size="small"
													variant="remove"
													shortPadding
													name={`who_develop[${index}]_remove_button`}
													onClick={(event) => {
														event.preventDefault();
														remove(index);
													}}
												>
													<FaMinus />
												</CircularButton>
											</Column>
										</Row>
									);
								}}
								// eslint-disable-next-line no-shadow
								endComponent={({ append, emptyValue }) => {
									return (
										<Row>
											<Column>
												<CircularButton
													right
													variant="info"
													color="white"
													name="who_develop_add_button"
													onClick={(event) => {
														event.preventDefault();
														append(emptyValue);
													}}
												>
													<FaPlus />
												</CircularButton>
											</Column>
										</Row>
									);
								}}
							/>
						</Column>
					</Row>
					<Row>
						<Column>
							<Title>Onde a Tecnologia pode ser aplicada?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<SelectField
								form={form}
								label="Região"
								name="where_can_be_applied"
								placeholder="Escolha uma região"
								validation={{ required: true }}
								options={[
									{
										value: 'semiarido',
										label: 'Semiárido',
									},
								]}
							/>
						</Column>
					</Row>

					<Row>
						<Column>
							<Title>Onde a Tecnologia já está implantada?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<PlacesAutocomplete
								value={addressInput}
								onChange={(value) => setAddressInput(value)}
								onSelect={async (value, options, properties) => {
									const toBePushed = properties;
									const response = await geocodeByPlaceId(toBePushed.placeId);
									if (response) {
										toBePushed.location = {
											lat: response[0].geometry.location.lat(),
											lng: response[0].geometry.location.lng(),
										};
									}
									setAddress([...address, toBePushed]);
								}}
							>
								{({
									getInputProps,
									suggestions,
									getSuggestionItemProps,
									loading,
								}) => (
									<div>
										<InputField
											{...getInputProps({
												placeholder: 'Procurar localidades...',
												className: 'location-search-input',
											})}
											form={{ register: () => {} }}
										/>
										<div className="autocomplete-dropdown-container">
											{loading && (
												<GoogleAddressSugestions>
													Loading...
												</GoogleAddressSugestions>
											)}
											<GoogleAddressSugestions>
												{suggestions.map((suggestion) => {
													const style = suggestion.active
														? {
																backgroundColor: '#fafafa',
														  }
														: {
																backgroundColor: '#fff',
														  };
													return (
														<div
															{...getSuggestionItemProps(suggestion, {
																style,
															})}
														>
															<Suggestion>
																{suggestion.description}
															</Suggestion>
														</div>
													);
												})}
											</GoogleAddressSugestions>
										</div>
									</div>
								)}
							</PlacesAutocomplete>
						</Column>
					</Row>
					<Row>
						<Column>
							{address.map((element, index) => {
								return (
									<IconRow row>
										<IconLink>
											<FaMapMarkerAlt size="2rem" /> {element.description}
										</IconLink>
										<CircularButton
											variant="remove"
											height="2"
											width="2"
											onClick={() => deleteAlreadyImpletedMarker(index)}
										>
											<FaTrash size="1em" />
										</CircularButton>
									</IconRow>
								);
							})}
						</Column>
					</Row>
				</Column>
				<Column>
					<Title>Fotos e Vídeos da tecnologia</Title>

					<Dropzone
						accept="image/*"
						onDrop={(acceptedFiles) => onDropImgs(acceptedFiles)}
						ref={imgsRef}
					>
						{({ getRootProps, getInputProps }) => (
							<UploadBox {...getRootProps()}>
								<input {...getInputProps()} />
								<SvgBox>
									<IconBox>
										<FaFileUpload size="5rem" />
										<IconTextBox>
											<Div>Upload</Div>
											<Div>1280px x 720px</Div>
										</IconTextBox>
									</IconBox>
								</SvgBox>
							</UploadBox>
						)}
					</Dropzone>
					<UploadedImages>
						{previewedImgFiles.map((element, index) => {
							return (
								<IconRow>
									<Media key={element.src} src={URL.createObjectURL(element)} />
									<CircularButton
										variant="remove"
										height="3"
										width="3"
										onClick={() => deleteMedia(index)}
									>
										<FaTrash size="1.5em" />
									</CircularButton>
								</IconRow>
							);
						})}
					</UploadedImages>

					<Title>Documentos</Title>
					<Dropzone
						accept=".pdf"
						onDrop={(acceptedFiles) => onDropPdfs(acceptedFiles)}
						ref={pdfRef}
					>
						{({ getRootProps, getInputProps }) => (
							<UploadBox {...getRootProps()}>
								<input {...getInputProps()} />
								<SvgBox>
									<IconBox>
										<FaFileUpload size="5rem" />
										<IconTextBox>
											<Div>Upload</Div>
											<Div>.PDF</Div>
										</IconTextBox>
									</IconBox>
								</SvgBox>
							</UploadBox>
						)}
					</Dropzone>
					<UploadedDocuments>
						{previewedPdfFiles.map((element, index) => {
							return (
								<IconRow row>
									<IconLink>
										<FaFilePdf size="2rem" /> {element.name}
									</IconLink>
									<CircularButton
										variant="remove"
										height="2"
										width="2"
										onClick={() => deleteAttachment(index)}
									>
										<FaTrash size="1em" />
									</CircularButton>
								</IconRow>
							);
						})}
					</UploadedDocuments>
				</Column>
			</Row>
		</Wrapper>
	);
};

MapAndAttachments.propTypes = {
	form: PropTypes.shape({}),
};

MapAndAttachments.defaultProps = {
	form: {},
};

export default MapAndAttachments;
