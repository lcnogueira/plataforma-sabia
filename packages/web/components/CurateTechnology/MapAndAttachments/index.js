import React from 'react';

import { FaFilePdf } from 'react-icons/fa';
import Videos from '../../Technology/Details/Videos';
import Geolocation from '../../Technology/Details/Tabs/GeoLocation';
import { useTechnology, useModal } from '../../../hooks';
import { Container, ContentBox } from '../styles';
import * as S from './styles';

const MapAndAttachments = () => {
	const { technology } = useTechnology();
	const { openModal } = useModal();

	return (
		<Container>
			<ContentBox flexBasis="100%">
				<Geolocation />

				<S.UploadsTitle>Fotos da Tecnologia</S.UploadsTitle>
				{technology.attachments?.images?.length ? (
					<S.UploadedImages>
						{technology.attachments?.images.map((element, index) => (
							<S.IconRow key={element.url}>
								<S.Media
									src={element.url}
									role="button"
									aria-label="Open image gallery"
									onClick={() => {
										openModal(
											'imagesGallery',
											{
												technology,
												settings: {
													arrows: false,
													dots: true,
													initialSlide: index,
												},
											},
											{ customModal: true },
										);
									}}
								/>
								{element.id === technology.thumbnail_id && (
									<S.ImageCaption>Imagem de capa</S.ImageCaption>
								)}
							</S.IconRow>
						))}
					</S.UploadedImages>
				) : (
					<p>Nenhuma foto cadastrada</p>
				)}

				<S.UploadsTitle>Vídeos da Tecnologia</S.UploadsTitle>
				{technology.videos?.length ? (
					<Videos data={technology.videos} />
				) : (
					<p>Nenhum vídeo cadastrado</p>
				)}

				<S.UploadsTitle>Documentos</S.UploadsTitle>
				{technology.attachments.documents?.length ? (
					<S.UploadedDocuments>
						{technology.attachments.documents.map((element) => (
							<S.IconRow row key={element.url}>
								<S.IconLink href={element.url}>
									<FaFilePdf size="2rem" /> {element.filename}
								</S.IconLink>
							</S.IconRow>
						))}
					</S.UploadedDocuments>
				) : (
					<p>Nenhum documento cadastrado</p>
				)}
			</ContentBox>
		</Container>
	);
};

export default MapAndAttachments;
