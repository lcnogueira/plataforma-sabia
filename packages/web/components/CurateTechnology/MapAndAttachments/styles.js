import styled, { css } from 'styled-components';
import * as MapAndAttachments from '../../TechnologyForm/MapAndAttachments/styles';

export const UploadsTitle = styled.span`
	${({ theme: { colors } }) => css`
		display: block;
		font-weight: 500;
		font-size: 1.4rem;
		margin-bottom: 1rem;
		text-transform: uppercase;
		color: ${colors.lightGray};

		&:not(:first-child) {
			margin-top: 2.4rem;
		}
	`}
`;

export const UploadedDocuments = styled(MapAndAttachments.UploadedDocuments)``;

export const UploadedImages = styled(MapAndAttachments.UploadedImages)`
	grid-template-columns: repeat(auto-fill, 20%);
`;

export const Media = styled(MapAndAttachments.Media)`
	cursor: pointer;
	filter: opacity(1);
`;

export const IconRow = styled(MapAndAttachments.IconRow)``;

export const IconLink = styled(MapAndAttachments.IconLink)``;

export const ImageCaption = styled.p`
	${({ theme: { colors } }) => css`
		font-size: 1.2rem;
		margin-top: -1rem;
		color: ${colors.lightGray};
	`}
`;
