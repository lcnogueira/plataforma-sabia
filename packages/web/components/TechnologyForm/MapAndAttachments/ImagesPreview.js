import React from 'react';
import { FiTrash2, FiImage } from 'react-icons/fi';

import { IconRow, Media, Button, RadioWrapper } from './styles';

const ImagesPreview = ({ previewedImgFiles, value, onChange, deleteAttachment }) => {
	return (
		previewedImgFiles?.map((element, index) => (
			<IconRow key={element.url} alignItems="flex-start">
				<RadioWrapper>
					<input
						id={element.url}
						type="radio"
						name="thumbnail_id"
						value={element.id}
						defaultChecked={value === element.id || !index}
						onChange={onChange}
					/>
					{/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
					<label htmlFor={element.url}>
						<FiImage fontSize="1.6rem" />
						Usar como capa
					</label>
				</RadioWrapper>
				<Button
					onClick={() =>
						deleteAttachment({
							index,
							element,
							type: 'img',
						})
					}
				>
					<FiTrash2 fontSize="1.6rem" />
					Excluir
				</Button>
				<Media key={element.url} src={element.url} />
			</IconRow>
		)) || null
	);
};

export default ImagesPreview;
