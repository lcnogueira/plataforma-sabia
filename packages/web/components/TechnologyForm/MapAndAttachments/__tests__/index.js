import React from 'react';
import { render, screen, waitFor } from 'test-utils';
import MapAndAttachments from '..';
import { Form } from '../../../Form';

jest.mock('react-places-autocomplete', () => {
	return {
		__esModule: true,
		geocodeByPlaceId: () => {
			return <vid>geocodeByPlaceId</vid>;
		},
		default: () => {
			return <div>PlacesAutocomplete</div>;
		},
	};
});

const onSubmit = jest.fn(() => {});

test('it render the MapAndAttachments page', async () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MapAndAttachments />
		</Form>,
	);

	expect(container).toMatchSnapshot();
});

/* Videos */
test('should render the videos list', async () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MapAndAttachments />
		</Form>,
	);

	await waitFor(() => screen.getByText('Videos da tecnologia'));

	expect(container).toMatchSnapshot();
});

test('should the videos list is empty', async () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MapAndAttachments />
		</Form>,
	);

	await waitFor(() => screen.getByText('Nenhum v√≠deo adicionado'));

	expect(container).toMatchSnapshot();
});
