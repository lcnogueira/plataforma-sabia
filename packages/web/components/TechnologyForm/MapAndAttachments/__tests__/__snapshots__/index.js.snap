// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the MapAndAttachments page 1`] = `
.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: hsla(4,85%,57%);
  margin-left: 0.5rem;
  display: inline-block;
}

.c8 {
  width: 100%;
  margin: 0.5rem 0;
  font-size: 1.4rem;
}

.c8 .react-select__control {
  min-height: 4.4rem;
  border-radius: 0.2rem;
}

.c8 .react-select__value-container > div {
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 0rem;
  margin-bottom: 1rem;
  background-color: transparent;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  margin-bottom: 4rem;
}

.c9 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  margin-bottom: 1rem;
  border: 1px solid hsl(0,0%,74%);
  padding: 1rem;
  cursor: pointer;
}

.c9:hover {
  background-color: hsl(0,0%,74%);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 > * {
  margin: 0.25rem;
}

.c12 {
  font-size: 1.2rem;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 {
  display: grid;
  grid-template-columns: repeat(auto-fill,30%);
  grid-gap: 1rem;
}

.c10 {
  text-align: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: auto 0;
}

@media (max-width:1170px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (max-width:450px) {
  .c13 {
    grid-template-columns: repeat(auto-fill,100%);
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia é desenvolvida?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div>
                PlacesAutocomplete
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            />
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia pode ser aplicada?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <label
                  class="c5"
                  for="terms.where_can_be_applied"
                >
                  Região
                  <span
                    aria-label="Indicador visual de campo obrigatório"
                    class="c6"
                  >
                    *
                  </span>
                </label>
                <div
                  class="c7"
                >
                  <div
                    class="c8 react-select-container css-2b097c-container"
                    id="where_can_be_applied"
                  >
                    <div
                      class="react-select__control css-yk16xz-control"
                    >
                      <div
                        class="react-select__value-container css-g1d714-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-1wa3eu0-placeholder"
                        >
                          Escolha uma região
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class="react-select__input"
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-label="Região"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-2-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: 1.4rem; font-weight: 400; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-6q0nyr-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="terms.where_can_be_applied"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia já está implantada?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div>
                PlacesAutocomplete
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            />
          </div>
        </div>
        <div
          class="c3"
        >
          <h3
            class=""
          >
            Fotos e Vídeos da tecnologia
          </h3>
          <div
            class="c9"
            tabindex="0"
          >
            <input
              accept="image/*"
              autocomplete="off"
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                <svg
                  fill="currentColor"
                  height="5rem"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="5rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
                  />
                </svg>
                <div
                  class="c12"
                >
                  <div
                    class=""
                  >
                    Upload
                  </div>
                  <div
                    class=""
                  >
                    1280px x 720px
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c13"
          />
          <h3
            class=""
          >
            Documentos
          </h3>
          <div
            class="c9"
            tabindex="0"
          >
            <input
              accept=".pdf"
              autocomplete="off"
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                <svg
                  fill="currentColor"
                  height="5rem"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="5rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
                  />
                </svg>
                <div
                  class="c12"
                >
                  <div
                    class=""
                  >
                    Upload
                  </div>
                  <div
                    class=""
                  >
                    .PDF
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c14"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
